go-controller/pkg/node/node_ip_handler_linux.go

```diff
@@ -208,14 +208,16 @@ func (c *addressManager) runInternal(stopChan <-chan struct{}, doneWg *sync.Wait
 
 // updates OVN's EncapIP if the node IP changed
 func (c *addressManager) handleNodePrimaryAddrChange() {
+	c.Lock()
```

Comment:
I am confused - i see within `c.nodePrimaryAddrChanged()` that we get the lock again. Doesnt this cause deadlock?

Reply:
I _thought_ our linters would catch that case.